contracts/contracts/MerkleDrop.sol

```diff
@@ -77,11 +77,12 @@ contract MerkleDrop {
         }
 
         // The amount of tokens that should be held within the contract after burning
-        uint decayedRemainingValue = remainingValue - decay(remainingValue, now - decayStartTime, decayDurationInSeconds);
+        uint targetBalance = decayedEntitlementAtTime(remainingValue, now);
 
-        assert(decayedRemainingValue + spentTokens <= initialBalance);
+        assert(targetBalance + spentTokens <= initialBalance);
```

Comment:
Please change this to
```
assert(targetBalance <= currentBalance)
```
and move the currentBalance assignment before this assert.